File: Runtime/TransitionPatch.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

''' <summary>
''' Transition patch
''' </summary>
''' <remarks></remarks>
Class TransitionPatch

    Private m_Size As Double
    Private m_SeedCell As Cell
    Private m_EdgeCells As New Dictionary(Of Integer, Cell)()
    Private m_AllCells As New Dictionary(Of Integer, Cell)()

    Public Sub New(ByVal seedCell As Cell)
        Me.m_SeedCell = seedCell
    End Sub

    Public Property Size As Double
        Get
            Return Me.m_Size
        End Get
        Set(value As Double)
            Me.m_Size = value
        End Set
    End Property

    Public ReadOnly Property SeedCell As Cell
        Get
            Return Me.m_SeedCell
        End Get
    End Property

    Public ReadOnly Property EdgeCells As Dictionary(Of Integer, Cell)
        Get
            Return Me.m_EdgeCells
        End Get
    End Property

    Public ReadOnly Property AllCells As Dictionary(Of Integer, Cell)
        Get
            Return Me.m_AllCells
        End Get
    End Property

End Class

File: Runtime/TransitionPatch.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.Generic;

namespace SyncroSim.STSim
{
    internal class TransitionPatch
    {
        private double m_Size;
        private Cell m_SeedCell;
        private Dictionary<int, Cell> m_EdgeCells = new Dictionary<int, Cell>();
        private Dictionary<int, Cell> m_AllCells = new Dictionary<int, Cell>();

        public TransitionPatch(Cell seedCell)
        {
            this.m_SeedCell = seedCell;
        }

        public double Size
        {
            get
            {
                return this.m_Size;
            }
            set
            {
                this.m_Size = value;
            }
        }

        public Cell SeedCell
        {
            get
            {
                return this.m_SeedCell;
            }
        }

        public Dictionary<int, Cell> EdgeCells
        {
            get
            {
                return this.m_EdgeCells;
            }
        }

        public Dictionary<int, Cell> AllCells
        {
            get
            {
                return this.m_AllCells;
            }
        }
    }
}
